includes/connection.php

```diff
@@ -188,6 +205,35 @@ function remove_connection( $user_id ) {
 add_action( 'airstory_user_disconnect', __NAMESPACE__ . '\remove_connection' );
 
 /**
+ * Given a user ID and an array of blog IDs, connect or disconnect users from Airstory.
+ *
+ * @param int   $user_id  The user ID to update.
+ * @param array $blog_ids An array of blog IDs where the user should be connected. Any blog *not*
+ *                        in the array will be disconnected.
+ */
+function set_connected_blogs( $user_id, $blog_ids ) {
+	if ( ! is_multisite() ) {
+		return;
+	}
+
+	$available_blogs = Settings\get_available_blogs( $user_id );
```

Comment:
and if they only have one, you can just handle the add / remove connection part without a loop